cpuid/src/cpu_leaf/leaf_0x1/engine.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::registers::*;
+use bit_helper::BitHelper;
+use cpu_leaf::*;
+use kvm_bindings::kvm_cpuid_entry2;
+
+// CPUID bits in ebx, ecx, and edx.
+const EBX_CLFLUSH_CACHELINE: u32 = 8;
+
+/// Sets leaf 01H EBX[23-16].
+///
+/// The maximum number of addressable logical CPUs is computed as the closest power of 2
+/// higher or equal to the CPU count configured by the user.
+pub fn get_max_addressable_lprocessors(cpu_count: u8) -> Result<u8, Error> {
+    let mut max_addressable_lcpu = (cpu_count as f64).log2().ceil();
+    max_addressable_lcpu = (2 as f64).powf(max_addressable_lcpu);
+    // check that this number is still an u8
+    if max_addressable_lcpu > u8::max_value().into() {
+        return Err(Error::VcpuCountOverflow);
+    }
+    Ok(max_addressable_lcpu as u8)
+}
+
+pub struct Engine {
```

Comment:
What is Engine suppose to be?

Reply:
LeafEngine.process is supposed to be the handler that configures each leaf. There was a switch-case in filter_cpuid that used to do that.

Engine is just an implementation of LeafEngine